rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -390,35 +378,120 @@
         .attr("y2", ChartSettings.height);
     }
 
+    /**
+     * millis is the time between current time and the time of last fetch (according to browser), plus
+     * the time between the last fetch and the run creation (according to server)
+     * @returns {number}
+     */
+    function getSecondsSinceRunOpened() {
+      var openRun = _.findWhere(state.runs, {open: true});
+
+      // It should almost always be true that if state.is_open is true that openRun exists
+      // But there is a small delay between when state.is_open is set and when the runs
+      // are refreshed from the server
+      if (openRun) {
+        var millis = moment().diff(state.lastPoll) + moment(state.server_now).diff(moment(openRun.created));
+        return Math.floor(millis / 1000);
+      }
+      return 0;
+    }
+
+    /**
+     * Render buttons and select element above the chart
+     */
+    function renderButtons() {
+      var $buttonsRow = $element.find(buttonsRowSel);
+      var $problemButton = $element.find(problemStatusBtnSel);
+      var $timer = $element.find(timerSel);
+      var $timerSpinner = $element.find(timerSpinnerSel);
+      var $timerSpinnerText = $element.find(timerSpinnerTextSel);
+
+      if (state.selectedRuns.length !== 1) {
+        // Don't show these buttons for the the compare view
+        $buttonsRow.toggleClass('hidden', true);
+        return;
+      }
+
+      $buttonsRow.toggleClass('hidden', false);
+      $problemButton.text((state.is_open ? "Close" : "Open") + " Problem Now");
+
+      var totalSeconds = getSecondsSinceRunOpened();
+      var pollSeconds = 0, pollMinutes = 0;
+      if (state.is_open) {
+        pollSeconds = totalSeconds % 60;
+        pollMinutes = Math.floor(totalSeconds / 60);
+      }
+      $timer.text(pollMinutes + "m : " + pollSeconds + "s");
+      $timerSpinner.toggleClass('hidden', !state.is_open);
+      $timerSpinnerText.toggleClass('hidden', !state.is_open);
+    }
+
+    /**
+     * Update the timer every second
+     */
+    function updateTimer() {
+      if (state.is_open) {
+        setTimeout(updateTimer, TIMER_MILLIS);
+      }
+      render();
+    }
+
     /**
      * Read from the responses API and put the new value in the rendering state.
      * If the problem is open, schedule another poll using this function.
      */
     function pollForResponses() {
-      $.get(responsesUrl).then(function(newState) {
-        state = _.assign({}, state, newState);
+      if (state.is_open) {
+        setTimeout(pollForResponses, POLLING_MILLIS);
+      }
+
+      // make sure this updates at least once a second
+      if (state.isPolling) {
+        // API call is still in progress
+        return;
+      }
 
+      state.isPolling = true;
+      $.get(responsesUrl).then(function(newState) {
+        state = _.assign({}, state, newState, {
+          isPolling: false,
+          lastPoll: moment()
+        });
```

Comment:
I was going to say that we only need to render the chart here, but maybe it's best that we leave as-is and do a full render to protect against cases where the problem open status was changed in another tab or something.